### PR TITLE
add per lecturer todo view

### LINKED ISSUE #137: Add a "per lecturer todo" view
Thomas Schulz created a table which looked like this

- Lecturer 1, Delegates: A, B
       - Lecture 1: status
       - Lecture 2: status

- Lecturer 2 and so on.

which obviously is quite helpful for knowing which lecturer needs a friendly reminder to approve his lectures.

### PR DESCRIPTION
**Pull request by janno42**
fix #137

### CODE REVIEW

File: evap/staff/templates/staff_semester_todo.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                            <th style="width: 15%">{% trans "Status" %}</th>
+                            <th style="width: 20%">{% trans "Evaluation Period" %}</th>
+                            <th style="width: 25%">{% trans "Last modified by" %}</th>
+                       </tr>
+                    </thead>
+                    <tbody>
+                        {% for course in responsible_entry.courses %}
+                            <tr
+                                {%  if course.state != "prepared" %}
+                                 class="deactivate"
```
Comment (karyon) [inline evap/staff/templates/staff_semester_todo.html]: indentation

Comment (janno42) [inline evap/staff/templates/staff_semester_todo.html]: fixed

File: evap/staff/views.py [hunk 2]
```diff
@@ -275,6 +275,24 @@ def semester_lottery(request, semester_id):
...
+    semester = get_object_or_404(Semester, id=semester_id)
+
+    responsibles = set()
+    prepared_courses = semester.course_set.filter(state__in=['prepared']).all()
+    courses = semester.course_set.filter(state__in=['prepared', 'lecturerApproved']).all()
+
+    for course in prepared_courses:
+        responsibles.add(course.responsible_contributor)
+
+    responsibles = list(responsibles)
```
Comment (karyon) [inline evap/staff/views.py]: `responsibles = list(set(course.responsible_contributor for course in prepared_courses))`?

or
```
responsibles = (course.responsible_contributor for course in prepared_courses)
responsibles = list(set(responsibles))
```
to be more explicit

File: evap/staff/views.py [hunk 3]
```diff
@@ -275,6 +275,24 @@ def semester_lottery(request, semester_id):
...
+    responsibles.sort(key = lambda responsible: (responsible.last_name, responsible.first_name))
+
+    responsible_list = [dict(responsible=responsible, courses=[course for course in courses if course.responsible_contributor.id == responsible.id], delegates=responsible.delegates.all()) for responsible in responsibles]
```
Comment (karyon) [inline evap/staff/views.py]: you sort the list but then create an unsorted dict

Comment (janno42) [inline evap/staff/views.py]: well the dict contains a sorted list, doesn't it?

Comment (janno42) [inline evap/staff/views.py]: ah, okay. well that's nonsense, right.

File: evap/staff/views.py [hunk 4]
```diff
@@ -275,6 +275,24 @@ def semester_lottery(request, semester_id):
...
+
+    template_data = dict(semester=semester, responsible_list=responsible_list)
+    return render(request, "staff_semester_todo.html", template_data)
```
Comment (karyon) [inline evap/staff/views.py]: is it correct and intended that the page displays approved courses only from contributors that have prepared courses? why not show them for all or no contributors?

File: evap/staff/views.py (hunk 3 above)
Comment (janno42) [inline evap/staff/views.py]: okay. again. there is a list where i add dictionaries for each of the responsibles. they are sorted so the resulting *list* (which contains a number of *dict*s) will be sorted by responsible.

Comment (karyon) [inline evap/staff/views.py]: maybe you actually want a list of tuples here?

File: evap/staff/views.py (hunk 4 above)
Comment (janno42) [inline evap/staff/views.py]: it is intended. approved courses are not necessary for the todo, so they are usually not shown. but i need to know who approved these courses if the responsible lecturer has some not approved courses left to know to whom i have to talk.

**Comment by karyon** [comment]
:+1:

**Pull request merged**